modules/shredder/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/shredder/batch/ShredJob.scala

```diff
@@ -140,36 +140,44 @@ class ShredJob(@transient val spark: SparkSession,
         timestampsAccumulator.add(event)
         val isSyntheticDupe = syntheticDupesBroadcasted.value.contains(event.event_id)
         val withDupeContext = if (isSyntheticDupe) Deduplication.withSynthetic(event) else event
-        Shredded.fromEvent[Id](IgluSingleton.get(igluConfig), isTabular, atomicLengths, ShredJob.BadRowsProcessor)(withDupeContext).value
+        withDupeContext
       }
     }.cache()
 
-    // Update the shredded JSONs with the new deduplicated event IDs and stringify
-    val shreddedData = shredded.flatMap {
-      case Right(shredded) => shredded
-      case Left(row) => List(Shredded.fromBadRow(row))
+    val (isGoodEmpty, goodCount, bad) = if (config.output.wideRow) {
+      val bad = eventsWithDupeContext.flatMap(_.swap.toOption)
+      val wideRowed = eventsWithDupeContext.flatMap(_.toOption)
+        .map(_.toJson(true).noSpaces)
+      Sink.writeWideRowed(spark, config.output.compression, wideRowed, outFolder.append(ShredJob.WideRowFolderPrefix))
+      (wideRowed.isEmpty, wideRowed.countAsync(), bad)
+    } else {
+      val shredded = eventsWithDupeContext.map { e =>
+        e.flatMap(Shredded.fromEvent[Id](IgluSingleton.get(igluConfig), isTabular, atomicLengths, ShredJob.BadRowsProcessor)(_).value)
+      }.cache()
+      val shreddedGood = shredded.flatMap(_.toOption).flatMap(identity)
+      val shreddedBad = shredded.flatMap(_.swap.toOption)
+
+      Sink.writeShredded(spark, config.output.compression, shreddedGood.flatMap(_.tsv), outFolder)
+      Sink.writeShredded(spark, config.output.compression, shreddedGood.flatMap(_.json), outFolder)
+
+      (shreddedGood.isEmpty, shreddedGood.countAsync(), shreddedBad)
     }
 
-    val goodCount: Future[Long] = shredded.filter(_.isRight).countAsync()
-
     // Data that failed TSV transformation
-    val shreddedBad = common.flatMap(_.swap.toOption.map(Shredded.fromBadRow).flatMap(_.json))
-
-    // Final output
-    Sink.writeShredded(spark, config.output.compression, shreddedData.flatMap(_.tsv), outFolder)
-    Sink.writeShredded(spark, config.output.compression, shreddedData.flatMap(_.json) ++ shreddedBad, outFolder)
+    val shreddedBad = (common.flatMap(_.swap.toOption) ++ bad).map(Shredded.fromBadRow).flatMap(_.json)
+    Sink.writeShredded(spark, config.output.compression, shreddedBad, outFolder)
 
     val shreddedTypes = shreddedTypesAccumulator.value.toList
     val batchTimestamps = timestampsAccumulator.value
     val timestamps = Timestamps(jobStarted, Instant.now(), batchTimestamps.map(_.min), batchTimestamps.map(_.max))
 
-    val isEmpty = batchTimestamps.isEmpty && shreddedTypes.isEmpty && shreddedData.isEmpty()  // RDD.isEmpty called as last resort
+    val isEmpty = batchTimestamps.isEmpty && shreddedTypes.isEmpty && isGoodEmpty  // RDD.isEmpty called as last resort
     val finalShreddedTypes = if (isEmpty) Nil else ShreddedType(Common.AtomicSchema, Format.TSV) :: shreddedTypes
```

Comment:
I didn't change anything about returning shredded types yet. I just wanted to start a discussion first. 

So, in the Snowflake Loader, only [shred types](https://github.com/snowplow-incubator/snowplow-snowflake-loader/blob/0.8.2/transformer/src/main/scala/com/snowplowanalytics/snowflake/transformer/Transformer.scala#L47) are returned since there is no TSV format in there. With wide row, there will be no format distinction between TSV and JSON. Should we remove formatting part and just sent shredded types similar to Snowflake Loader ?

Reply:
> Should we remove formatting part and just sent shredded types similar to Snowflake Loader ?

I'd say - yes. Or perhaps, send it with something like `WIDEROW` (or bettter name) format.